src/plugins/explore/public/application/components/query_panel.tsx

```diff
@@ -223,50 +156,54 @@ export const QueryPanel: React.FC<QueryPanelProps> = ({ datePickerRef }) => {
       }
 
       try {
+        // Get the effective language for autocomplete (PPL -> PPL_Simplified for explore app)
+        const effectiveLanguage = getEffectiveLanguageForAutoComplete(queryLanguage, 'explore');
+
         // Use centralized IndexPattern from context
         const suggestions = await services?.data?.autocomplete?.getQuerySuggestions({
           query: editorRef.current?.getValue() ?? '',
           selectionStart: model.getOffsetAt(position),
           selectionEnd: model.getOffsetAt(position),
-          language: queryLanguage,
-          indexPattern: contextIndexPattern,
+          language: effectiveLanguage,
+          indexPattern: indexPattern as any,
           datasetType: dataset?.type,
           position,
-          services: services as any, // Type cast for compatibility
+          services: services as any, // ExploreServices now includes appName, compatible with IDataPluginServices
```

Comment:
we have normally been enforcing good typing, but this file is going to be removed anyways after the new query panel is integrated, right?

Reply:
will fix type in a later PR together